Resources/public/js/views/ez-usermenuview.js

```diff
@@ -0,0 +1,193 @@
+         * @public
+         */
+        toggleDisplayed: function () {
+            this._set('displayed', !this.get('displayed'));
+        },
+
+        /**
+         * Toggles the visibility of a user menu.
+         *
+         * @method _uiToggleUserMenu
+         * @protected
+         * @param event {Object} event facade
+         */
+        _uiToggleUserMenu: function (event) {
+            var userMenu = this.get('container');
+
+            if (event.newVal) {
+                userMenu.removeClass(CLASS_HIDDEN);
+            } else {
+                userMenu.addClass(CLASS_HIDDEN);
+            }
+        },
+
+        /**
+         * Set the click outside event listener
+         *
+         * @method _setClickOutsideUserMenuHandler
+         * @protected
+         * @param event {Object} event facade
+         */
+        _setClickOutsideUserMenuHandler: function (event) {
+            if (event.newVal) {
+                this._clickOutsideUserMenuHandler = this.get('container').on('clickoutside', this._hideMenuOnClickOutside, this);
+            } else {
+                this._clickOutsideUserMenuHandler.detach();
+            }
+
+            return this;
+        },
+
+        /**
+         * Checks if clicked outside the user menu container and hides a user menu if it is visible.
+         *
+         * @method _hideMenuOnClickOutside
+         * @protected
+         */
+        _hideMenuOnClickOutside: function (event) {
+            if (!event.target.hasClass(CLASS_USER_PROFILE) && event.target.ancestor(SELECTOR_USER_PROFILE) === null && this.get('displayed')) {
+                this.toggleDisplayed();
+            }
+        },
+
+        /**
+         * Sets active flag to the item view
+         *
+         * @method _setActiveViews
+         * @protected
+         * @param event {Object} event facade
+         */
+        _setActiveViews: function (event) {
+            var items = this.get('items');
+
+            items.forEach(function (item) {
+                item.set('active', event.newVal);
+            });
+        },
+
+        /**
+         * Renders menu items views
+         *
+         * @method _renderMenuItems
+         * @protected
+         */
+        _renderMenuItems: function () {
+            var menuItemsContainer = this.get('container'),
+                items = this.get('items');
+
+            items.sort(function (a, b) {
+                return b.get('priority') - a.get('priority');
+            });
+
+            items.forEach(Y.bind(function (item) {
+                menuItemsContainer.append(item.render().get('container'));
+            }, this));
+        },
+
+        /**
+         * Adds item to the menu
+         *
+         * @method addMenuItem
+         * @public
+         * @param view {Object}
+         */
+        addMenuItem: function (view) {
+            var items = this.get('items');
+
+            items.push(view);
+            this.set('items', items);
```

Comment:
nitpick: actually you don't have to do that, the array is updated by the previous line and this array is referenced in the `items` attribute.